### PR TITLE
WIP: AEROGEAR-2244 security dashboard revised

### PR DESCRIPTION
**Pull request by aliok**
This is a WIP.

Closing https://github.com/aerogearcatalog/metrics-apb/pull/42 as I have created a mess and pushing more commits on top of that branch makes things ugly.

Commits: 
* https://github.com/aerogearcatalog/metrics-apb/pull/44/commits/c4521d4bb966bc987ba91ffec91da14994a16a6e: Added a .gitignore file
* https://github.com/aerogearcatalog/metrics-apb/pull/44/commits/14824206a345d5a5aca0664511a273ca8dee1f93: Mobile services dashboard to have mobile security overview graphs

### CODE REVIEW

**Comment by aliok** [comment]
@paolobueno Do you mind checking the updates in mobile services dashboard. No need to wait for the separate dashboard to be pushed.

File: roles/provision-metrics-apb/files/mobile-services-dashboard.json [hunk 1]
```diff
@@ -1171,6 +1171,516 @@
...
+      },
+      "tableColumn": "count",
+      "targets": [
+        {
+          "$$hashKey": "object:351",
+          "alias": "",
+          "expr": "",
+          "format": "table",
+          "intervalFactor": 1,
+          "rawSql": "SELECT count(clientId)\nFROM mobileAppMetrics\nWHERE $__timeFilter(event_time)\n  AND data ? 'security';",
```
Comment (paolobueno) [inline roles/provision-metrics-apb/files/mobile-services-dashboard.json]: @aliok  Found a bug on this one, missing the distinct clause:
```sql
SELECT count(distinct clientId)
FROM mobileAppMetrics
WHERE $__timeFilter(event_time)
  AND data ? 'security';
```

File: roles/provision-metrics-apb/files/mobile-services-dashboard.json [hunk 2]
```diff
@@ -1171,6 +1171,516 @@
       ],
       "title": "Mobile app metrics",
       "type": "row"
+    },
+    {
+      "collapsed": false,
```
Comment (paolobueno) [inline roles/provision-metrics-apb/files/mobile-services-dashboard.json]: Others are `true`. Makes the security group be the only one open by default right now.

**Comment by paolobueno** [comment]
@aliok Revised the dashboard in the main one, just found a single bug above.

In general when you use `data->'key'` access it should be fine to leave out the corresponding `data ? 'key'` check in the `WHERE` clause, since it's done implicitly.
Also no undefined exceptions so `data->'app'->>'appId'` is safe without `WHERE data ? 'app'`.

**Comment by paolobueno** [comment]
I guess using the `?` would allow the queries to use jsonb indexes if they're added in the future, so I'm ok with keeping them too, just a little more verbose.

File: roles/provision-metrics-apb/files/mobile-services-dashboard.json (hunk 1 above)
Comment (aliok) [reply roles/provision-metrics-apb/files/mobile-services-dashboard.json]: good catch. will fix it

File: roles/provision-metrics-apb/files/mobile-services-dashboard.json (hunk 2 above)
Comment (aliok) [reply roles/provision-metrics-apb/files/mobile-services-dashboard.json]: +1

**Comment by aliok** [comment]
thanks @paolobueno . I will fix them and push along with the main security dashboard.

**Comment by aliok** [comment]
Closing in favor of https://github.com/aerogearcatalog/metrics-apb/pull/45

**Comment by aliok** [comment]
This PR is closed after the discussion in https://groups.google.com/forum/#!topic/aerogear/MgFcW2gF_oI

**Pull request closed without merging**